api/v1alpha4/azuremanagedcontrolplane_types.go

```diff
@@ -157,7 +157,7 @@ type LoadBalancerProfile struct {
 type APIServerAccessProfile struct {
 	// AuthorizedIPRanges - Authorized IP Ranges to kubernetes API server.
 	// +optional
-	AuthorizedIPRanges []string `json:"authorizedIPRanges,omitempty"`
+	AuthorizedIPRanges *[]string `json:"authorizedIPRanges,omitempty"`
```

Comment:
/hold 

we shouldn't change the older API versions in the main branch, those are only there for conversion purposes. Youll need to remove this change and handle the conversion to pointer between v1alpha4 and v1beta1. Side node: is it really necessary to change the API? could you achieve similar results without making this field a pointer, given that in go slices have a built-in nil value? 

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required